src/xrpld/app/tx/detail/SetTrust.cpp

```diff
@@ -182,6 +228,63 @@ SetTrust::preclaim(PreclaimContext const& ctx)
         }
     }
 
+    // Checking all freeze/deep freeze flag invariants.
+    auto sleRippleState =
+        ctx.view.read(keylet::line(id, uDstAccountID, currency));
+
+    if (sleRippleState)
+    {
+        if (ctx.view.rules().enabled(featureDeepFreeze))
+        {
```

Comment:
One problem that I didn't consider until I saw this addition: The trust line may not exist yet, but we still need to check that the flags are valid. It is possible to create a new trust line as frozen (or deep frozen). For example, an issuer may want to preemptively stop a known problematic account from receiving funds before the trust line exists.